tests/src/Kernel/ListPageRssControllerTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_list_pages\Kernel;
+
+use Drupal\Core\Url;
+use Drupal\node\Entity\NodeType;
+use Drupal\oe_list_pages\Controller\ListPageRssController;
+use Symfony\Component\DomCrawler\Crawler;
+
+/**
+ * Tests the List page RSS feeds.
+ */
+class ListPageRssControllerTest extends ListsEntityMetaTestBase {
+
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'language',
+    'oe_list_pages_event_subscriber_test',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+    $this->installConfig(['language']);
+    // Set up a current user to ensure anonymous users are available.
+    $this->setUpCurrentUser();
+  }
+
+  /**
+   * Test access to list page RSS route.
+   */
+  public function testAccess(): void {
+    $user = $this->createUser([], ['access content']);
+    // Create a node with list page metadata.
+    $list_page = $this->nodeStorage->create([
+      'type' => $this->nodeType->id(),
+      'title' => 'List Page',
+    ]);
+    $list_page->save();
+
+    // Create a content type without list page metadata.
+    $values = ['type' => 'article', 'name' => 'Article'];
+    $this->nodeType = NodeType::create($values);
+    $this->nodeType->save();
+    // Create a node without list page metadata.
+    $article = $this->nodeStorage->create([
+      'type' => 'article',
+      'title' => 'Article',
+    ]);
+    $article->save();
+
+    // Assert we can access an RSS route for a node that has list page
+    // metadata assigned to it.
+    $route = Url::fromRoute('entity.node.list_page_rss', ['node' => $list_page->id()]);
+    $this->assertTrue($route->access($user));
+
+    // Assert we can not access an RSS route for a node that does not have
+    // list page metadata assigned to it.
+    $route = Url::fromRoute('entity.node.list_page_rss', ['node' => $article->id()]);
+    $this->assertFalse($route->access($user));
+  }
+
+  /**
+   * Test access to list page RSS route.
+   */
+  public function testBuild(): void {
```

Comment:
I understand that a Kernel thing makes things faster etc, but we also need a functional test. You apply a bunch of caching which you will only be able to check there that it gets invalidated. You can leave the kernel as well.